### PR TITLE
Add interface includes to IlmBase and OpenEXR

### PR DESCRIPTION
**Pull request by hjmallon**
Thank you for all the great improvements to CMake in 2.4.0! I am having another stab at integrating OpenEXR with the hunter package manager. Hunter tends to be a real work out for CMake implementations so I might have a couple of little tweaks to post.

This one means that `INTERFACE_INCLUDE_DIRECTORIES` are populated correctly on the targets in the CMake `XConfig.cmake` files.

### CODE REVIEW

**Review by meshula — APPROVED** [review]
lgtm, reference: https://cmake.org/cmake/help/v3.13/command/install.html

File: IlmBase/config/LibraryDefine.cmake [hunk 1]
```diff
@@ -94,6 +94,7 @@ function(ILMBASE_DEFINE_LIBRARY libname)
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    INCLUDES DESTINATION include
```
Comment (kdt3rd) [inline IlmBase/config/LibraryDefine.cmake]: should this and the others below not be ${CMAKE_INSTALL_INCLUDEDIR} to match the other gnuinstalldirs we use? Yes, this defaults to include, but just to be pedantic, it should probably match the use of the gnuinstalldirs variables...

**Review by kdt3rd — CHANGES_REQUESTED** [review]

File: IlmBase/config/LibraryDefine.cmake (hunk 1 above)
Comment (hjmallon) [reply IlmBase/config/LibraryDefine.cmake]: That makes sense, rebased and changed

**Comment by cary-ilm** [comment]
@kdt3rd will straighten out his CLA and then merge this.

**Review by kdt3rd — APPROVED** [review]

**Comment by kdt3rd** [comment]
Thanks again, sorry for the delay getting this merged

**Pull request merged**